### DISCUSSION TITLE
"Mis Bloques" se guardan como imagen incorrectamente

### DISCUSSION THREAD

**Discussion by ghost** [ScratchAddons/ScratchAddons#5601] [Discussion]
Al guardar un script como imagen con el addon blocks2image [en las líneas 177-181](https://github.com/ScratchAddons/ScratchAddons/blob/9b145eb4e4623af9db72d9beb8a81bd10112c916/addons/blocks2image/userscript.js#L177) con el navegador Brave se obtiene incorrectamente la imagen 
![block_10_1_2023-21_37_39](https://user-images.githubusercontent.com/16025506/211810330-f98c37cd-142a-4c3e-ada9-80c0bcb666cc.png)

lo correcto
![block_10_1_2023-21_42_29](https://user-images.githubusercontent.com/16025506/211810424-9023360b-f374-4246-b79f-dd227d4cdf14.png)

en el SVG el atributo **data-shapes** contiene tres valores **"c-block c-1 hat"** generando que no se reconosca el valor "**hat**"

**Una solución es**
```
    let dataShapes = svgchild.getAttribute("data-shapes").split(" ");
    svgchild.setAttribute(
      "transform",
      `translate(0,${dataShapes.includes("hat") ? "40" : "0"}) ${isExportPNG ? "scale(2)" : ""}`
    );

```

**Comment by ghost** [comment]
referercia de línea [en las líneas 177-181](https://github.com/ScratchAddons/ScratchAddons/blob/9b145eb4e4623af9db72d9beb8a81bd10112c916/addons/blocks2image/userscript.js#L177)

La idea sugerida en el post anteriór hace que la imagen de los bloques **Eventos** no sea la mejor.

Probé con esto
```
    let dataShapes = svgchild.getAttribute("data-shapes");
    let translateY = 0; // blocks no hat
    if(dataShapes==="c-block c-1 hat"){
     translateY = 40; // for My block
    }
    if(dataShapes==="hat"){
      translateY = 33; // for Events
    }    
    svgchild.setAttribute(
      "transform",
      `translate(0,${translateY}) ${isExportPNG ? "scale(2)" : ""}`
    );
```
Ahora los resultados de las imágenes son las esperadas. 

[La línea 241](https://github.com/ScratchAddons/ScratchAddons/blob/9b145eb4e4623af9db72d9beb8a81bd10112c916/addons/blocks2image/userscript.js#L241) puede ser eliminada.

**Reply by WorldLanguages** [reply]
Podrias decirme si los cambios que hice en https://github.com/ScratchAddons/ScratchAddons/pull/5604/files son los correctos?

**Reply by ghost** [reply]
OK. Son correctos, los acabo de probar.

**Comment by ghost** [comment]
ante el problema mencionado con exportar a svg https://github.com/ScratchAddons/ScratchAddons/pull/5604#issuecomment-1380709194
modifico el código a

```
    let dataShapes = svgchild.getAttribute("data-shapes");
    let scale = 1;
    let translateY = 0;
    isExportPNG ? scale = 2 : scale = 1;
    if (dataShapes === "c-block c-1 hat") {
      translateY = 20;
    }
    if (dataShapes === "hat") {
      translateY = 16;    
    }
    svgchild.setAttribute(
      "transform",
      `translate(0,${scale * translateY}) scale(${scale})`
    );
```
y eliminar la línea 241
